### ISSUE TITLE
Make image emoji a configuration option in Thunderbird

### ISSUE THREAD

**Issue by mganss** [mganss/EmojiAddIn#20]

**Comment by fidergo-stephane-gourichon** [comment]
> "Make image emoji a configuration option in Thunderbird"

What do you mean?

**Comment by fidergo-stephane-gourichon** [comment]
Is this related to #14?

I expected this plugin to implement emoji insertion by adding a few Unicode characters in the mail. That's the whole point of emoji. Else why accepts their constraints? If we choose to implement embedding/attaching images, better attach arbitrary images instead of the limited emoji set.

You're pretty right to state on #14 that embedding an image consumes bandwidth. Implementing emoji as links to external resources (to read a private e-mail, WT?) as an alternative to data-uri feels to me like rearranging the deck chairs of the Titanic while it sinks.

Sorry if I'm off-topic, no offense intended. Was it what you're considering here? Your words "image emoji a configuration" feel like a hint, but too short to be sure. Thanks for clarifying.

**Comment by mganss** [comment]
Yeah, exactly what's mentioned in #14. This should be a configuration switch that optionally flips the current behavior, i.e. click for character, shift-click for image. Feel free to submit a PR.

If you hate image emoji then there's always the option to only ever write plaintext messages like in the good old days 😉 In TB you can make plaintext messages the default (and optionally write HTML mails by shift-clicking).

**Comment by mganss** [comment]
Also, it would be cool to optionally show only platform emoji characters in the side pane. This will result in missing glyphs on many platforms but maybe these can be (optionally) filtered out using some glyph detection hackery.

**Comment by mganss** [comment]
I've added an option to insert characters instead of images by default. This option reverses the behavior, i.e. you get characters with a regular click and images with shift-click.
